### DISCUSSION TITLE
Future roadmap or product improvements?

### DISCUSSION THREAD

**Discussion by aman77707** [healthchecks/healthchecks#521] [Ideas]
Hi, 

I really like healthchecks as a product.
If you don't mind, I had 2 suggestions, though these are non trivial ones and may take some time to implement:

- Introducing Django Rest Framework(DRF) to handle APIs in the project
- Post DRF introduction, improving the front end to become a single page application using Modern Front End Frameworks(Using React, Angular, etc)

Do let me know your views about this. Also, I am ready to help, contribute and work my heart out for it to produce an initial prototype and later complete requirements. But, it is really important to know your opinion before I even think to proceed. 

If the above Idea doesn't seem nice, I would love to know if you have any future roadmap for the product.
Currently, I am studying this project and filling my knowledge gaps for healthechecks. I hope soon I should be able to start working on the open issues/backlogs and work with the community to resolve them.

Thanks,
Aman

**Comment by cuu508** [comment]
Hi Aman,

I don't have plans to move to SPA or modern front end frameworks. My plan is to keep it (as much as is practical) plain HTML/CSS. Submit data to the server using forms and HTTP POST. Healthchecks uses JS, but I try to limit its use. In a descending order of preference:

* no JS
* vanilla JS
* jQuery + libraries for specific tasks
* JS frameworks

I don't have a long term roadmap. For me it's mostly general maintenance and work on features that I get frequent requests for. No master plan behind it :-)

**Reply by aman77707** [reply]
Hi Pēteris, 

Thanks for letting me know your view. 
I hope I start contributing to healthchecks project to resolve the open issues and enhancements soon. :-)

Thanks.
Aman
